test_evm_eth_compliance/src/statetest/runner.rs

```diff
@@ -0,0 +1,377 @@
+    let account = create_accounts(&test_vm, 1, TokenAmount::from_whole(10_000))[0];
+    let mut pre_contract_cache: HashMap<B160, Address> = HashMap::new();
+
+    let timer = Instant::now();
+
+    for (name, unit) in suit.0.iter() {
+        // info!("{:#?}:{:#?}", name, unit);
+
+        // TODO :: Process env block, is it needed for FEVM context ?
+
+        // Process the "pre" block & deploy the contracts
+        for (_test_id, (address, info)) in unit.pre.iter().enumerate() {
+            assert!(
+                info.code.len() < MAX_CODE_SIZE,
+                "{}",
+                format!(
+                    "EVM byte code length ({}) is exceeding the maximum allowed of {MAX_CODE_SIZE}",
+                    info.code.len()
+                )
+            );
+
+            assert!(
+                info.code.first() != Some(&0xEF),
+                "EIP-3541: Contract code starting with the 0xEF byte is disallowed.",
+            );
+
+            let create_result = test_vm
+                .apply_message(
+                    account,
+                    EAM_ACTOR_ADDR,
+                    TokenAmount::zero(),
+                    fil_actor_eam::Method::CreateAccount as u64,
+                    Some(fil_actor_eam::InitAccountParams {
+                        eth_address: EthAddress(address.as_bytes().try_into().unwrap()),
+                    }),
+                )
+                .unwrap();
+
+            assert!(
+                create_result.code.is_success(),
+                "failed to create the new actor :: {:#?} | Path :: {:#?}",
+                create_result.message,
+                path,
+            );
+
+            let create_return: fil_actor_eam::Create2Return =
+                create_result.ret.deserialize().expect("failed to decode results");
+
+            let actor_address = Address::new_id(create_return.actor_id);
+
+            test_vm.mutate_state(actor_address, |st: &mut EvmState| {
+                let hasher = Code::try_from(SupportedHashes::Keccak256 as u64).unwrap(); // supported hashes are all implemented in multihash
+                let code_hash = multihash::Multihash::wrap(
+                    SupportedHashes::Keccak256 as u64,
+                    &hasher.digest(&info.code).to_bytes(),
+                )
+                .expect("failed to hash bytecode with keccak");
+
+                let cid = store
+                    .put(Code::Blake2b256, &Block::new(IPLD_RAW, info.code.clone()))
+                    .expect("failed to write bytecode");
+
+                let mut slots = StateKamt::new_with_config(store.clone(), KAMT_CONFIG.clone());
+
+                st.bytecode = cid;
+                st.bytecode_hash = code_hash;
+                st.contract_state = slots.flush().expect("failed to flush contract state");
+                st.nonce = info.nonce;
+            });
+
+            pre_contract_cache.insert(*address, actor_address);
+        }
```

Comment:
Can't you just call the constructor and override the nonce with a subsequent mutation?